pkg/policy/distillery/distillery.go

```diff
@@ -0,0 +1,184 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package distillery
+
+import (
+	"fmt"
+
+	identityPkg "github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/identity/cache"
+	"github.com/cilium/cilium/pkg/lock"
+	"github.com/cilium/cilium/pkg/policy"
+)
+
+var (
+	globalPolicyCache = newPolicyCache()
+)
+
+// SelectorPolicy represents a cached SelectorPolicy, previously resolved from
+// the policy repository and ready to be distilled against a set of identities
+// to compute datapath-level policy configuration.
+type SelectorPolicy interface {
+	// Consume returns the policy in terms of connectivity to peer
+	// Identities. The callee MUST NOT modify the returned pointer.
+	Consume(owner policy.PolicyOwner, cache cache.IdentityCache) *policy.EndpointPolicy
+}
+
+// policyCache represents a cache of resolved policies for identities.
+type policyCache struct {
+	lock.Mutex
+	policies map[*identityPkg.Identity]*cachedSelectorPolicy
```

Comment:
Does it matter that two Identity instances could describe the same (numeric) identity?

Reply:
(As discussed oflfine) Yes.

 I'll look at switching this over to `NumericIdentity`.